sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp

```diff
@@ -150,6 +150,20 @@ namespace Azure { namespace Core { namespace Http { namespace _internal {
               clientOptions.Transport));
     }
 
+    [[deprecated]]
+    explicit HttpPipeline(
+        Azure::Core::_internal::ClientOptions const& clientOptions,
+        std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>>&& perRetryPolicies,
+        std::vector<std::unique_ptr<Azure::Core::Http::Policies::HttpPolicy>>&& perCallPolicies)
+        : HttpPipeline(
+            clientOptions,
+            "security.attestation",
+            "1.0.0",
+            std::move(perRetryPolicies),
```

Comment:
To be sure, the best is to `git checkout` with all the GA tags and confirm that none of them was using that constuctor overload:
```
refs/tags/azure-core_1.0.0
refs/tags/azure-core_1.1.0
refs/tags/azure-core_1.2.0
refs/tags/azure-core_1.2.1
refs/tags/azure-core_1.3.0
refs/tags/azure-core_1.3.1
refs/tags/azure-core_1.4.0
refs/tags/azure-core_1.5.0
refs/tags/azure-core_1.6.0
refs/tags/azure-core_1.7.0
refs/tags/azure-core_1.7.1
refs/tags/azure-core_1.7.2
refs/tags/azure-identity_1.0.0
refs/tags/azure-identity_1.1.0
refs/tags/azure-identity_1.1.1
refs/tags/azure-identity_1.2.0
refs/tags/azure-identity_1.3.0
refs/tags/azure-security-keyvault-certificates_4.0.0
refs/tags/azure-security-keyvault-certificates_4.1.0
refs/tags/azure-security-keyvault-keys_4.0.0
refs/tags/azure-security-keyvault-keys_4.1.0
refs/tags/azure-security-keyvault-keys_4.2.0
refs/tags/azure-security-keyvault-keys_4.3.0
refs/tags/azure-security-keyvault-secrets_4.0.0
refs/tags/azure-security-keyvault-secrets_4.1.0
refs/tags/azure-storage-blobs_12.0.0
refs/tags/azure-storage-blobs_12.0.1
refs/tags/azure-storage-blobs_12.1.0
refs/tags/azure-storage-blobs_12.2.0
refs/tags/azure-storage-blobs_12.2.1
refs/tags/azure-storage-blobs_12.3.0
refs/tags/azure-storage-blobs_12.4.0
refs/tags/azure-storage-blobs_12.5.0
refs/tags/azure-storage-blobs_12.6.0
refs/tags/azure-storage-blobs_12.6.1
refs/tags/azure-storage-blobs_12.6.2
refs/tags/azure-storage-common_12.0.0
refs/tags/azure-storage-common_12.0.1
refs/tags/azure-storage-common_12.1.0
refs/tags/azure-storage-common_12.2.0
refs/tags/azure-storage-common_12.2.1
refs/tags/azure-storage-common_12.2.2
refs/tags/azure-storage-common_12.2.3
refs/tags/azure-storage-common_12.2.4
refs/tags/azure-storage-common_12.3.0
refs/tags/azure-storage-files-datalake_12.0.0
refs/tags/azure-storage-files-datalake_12.0.1
refs/tags/azure-storage-files-datalake_12.1.0
refs/tags/azure-storage-files-datalake_12.2.0
refs/tags/azure-storage-files-datalake_12.3.0
refs/tags/azure-storage-files-datalake_12.3.1
refs/tags/azure-storage-files-datalake_12.4.0
refs/tags/azure-storage-files-datalake_12.5.0
refs/tags/azure-storage-files-shares_12.0.0
refs/tags/azure-storage-files-shares_12.0.1
refs/tags/azure-storage-files-shares_12.1.0
refs/tags/azure-storage-files-shares_12.2.0
refs/tags/azure-storage-files-shares_12.2.1
refs/tags/azure-storage-files-shares_12.3.0
refs/tags/azure-storage-queues_12.0.0
```

Reply:
Attestation is the only service which was updated with distributed tracing support, thus it is the only service which used that constructor.

Reply:
I see, it was added only in this commit - https://github.com/Azure/azure-sdk-for-cpp/commit/99f45a5f18f20f44e87556c4f240c055c5e4c2d2

Yes, then I don't think it was ever used by anyone else other than Attestation.
I was thinking that the constructor was there since pre-historic times when we didn't have the TelemetryPolicy.